LiteCore/Storage/DataFile.cc

```diff
@@ -324,6 +324,19 @@ namespace litecore {
         }
     }
 
+    void DataFile::resetKeyStore(const string &name) {
+        logDebug("reset KVS '%s'", name.c_str());
+        auto i = _keyStores.find(name);
```

Comment:
I hate to say it, but … this isn't thread-safe 😩 I guess we need a mutex to protect `_keyStores`?

Reply:
Are there other non thread safe methods on here ? At a higher level this method is protected by the collections mutex, or are there other situations you are thinking of?